### DISCUSSION TITLE
Shallow prop with link component redirect to as prop value with app directory.

### DISCUSSION THREAD

**Discussion by sagarchoudhary96** [vercel/next.js#50274] [App Router]
I'm trying to implement https://vercel.com/blog/building-a-fast-animated-image-gallery-with-next-js this example with next13 and app directory. the link component is redirecting and reloading to the `as` props value.

**Comment by ciruz** [comment]
Shallow routing is not available with the App Router, only in `/pages` directory, see the API reference:

App Directory (`/app`): https://nextjs.org/docs/app/api-reference/components/link (no shallow)
Pages Router (`/pages`): https://nextjs.org/docs/pages/api-reference/components/link (shallow available).

Issues/Discussions:
https://github.com/vercel/next.js/discussions/48110
https://github.com/vercel/next.js/issues/48471
etc.

**Comment by sagarchoudhary96** [comment]
Does anyone have any updates/idea if it's gonna be implemented soon or there will be workaround for the same?
